### DISCUSSION TITLE
Why isn't there a Switch/Toggle component in Polaris?

### DISCUSSION THREAD

**Discussion by chloerice** [Shopify/polaris-react#7297] [General] [answered]
### Tl;dr: 
This question comes up at least a few times a year--last month it came up three times in one week 😅. The switch/toggle pattern has been and still is a hot-button topic in the design community and Shopify is no exception! The discussion it stirs up generally gets lost in Slack threads, along with the context unearthed from links shared to long forgotten GitHub issues and PRs. 

This discussion (and all Polaris discussions) is open for folks evolving the Shopify admin now and in the future to find the context and have the conversations they need to make great decisions quickly and build the best experience for merchants.

**Comment by chloerice** [comment]
## Related links

### GitHub Issues and PRs 
- https://github.com/Shopify/polaris/pull/600/files
- https://github.com/Shopify/polaris-styleguide/issues/1504 (Shopifolk access only)

### Specs and documentation
- https://www.w3.org/WAI/ARIA/apg/patterns/switch/

### Blog posts and articles
- https://web.dev/building-a-switch-component/

**Comment by dustinmalik** [comment, marked as answer]
Use a checkbox when the state is dirty like in settings. You have to click save to apply things.

Use a toggle when the API/save state is called immediately. For example turning on a flow in your store.

I believe there are toggles in other locations other than flow too.
